### PR TITLE
[docs] add from 6.x migration guide

### PR DESCRIPTION
**Pull request by cmark**

### CODE REVIEW

File: docs/migration/from-6x.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Migration from 6.x version
+
+The following major differences, features and topics are worth mentioning when comparing features present in Snow Owl 6 and 7 and migrating an existing 6.x deployment to Snow Owl 7.x.
+
+NOTE: It is highly recommended to keep the previous Snow Owl 6 deployment up and running until you have the data and all connected services migrated to the new version successfully. The new Snow Owl 7 system should get its own dedicated machine and deployment environment. Rolling back to the previous state should be available and must be executed when the upgrade cannot be performed successfully.
+
+## Java 11
+
+From Snow Owl 7.1, Snow Owl compiles and runs on Java 11+ versions. It is recommended to use the latest OpenJDK or OracleJDK 11 LTS version. Install from [here](https://openjdk.java.net/), [here](https://www.oracle.com/java/technologies/javase-downloads.html) or [here](https://adoptopenjdk.net/).
```
Comment (apeteri) [inline docs/migration/from-6x.md]: The links should have a more descriptive text to make it easier for the reader to figure out in advance where they lead. Oracle's Java SE now comes with commercial usage restrictions which might be worth mentioning here as well: https://www.oracle.com/technetwork/java/javase/overview/oracle-jdk-faqs.html

File: docs/migration/from-6x.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+3. Add the `classify:*` permission declaration and assign it to all roles that should be able to run classifications
+
+## Configuration changes
+
+Snow Owl 7 configuration file has been renamed to `snowowl.yml` (from `snowowl_config.yml`) and moved to the `<HOME>/configuration` folder.
+
+The following configuration settings have been changed:
+* `repository.database` configuration setting has been removed completely
+* `repository.numberOfWorkers` has been renamed to `repository.maxThreads` and its default value became `200`.
+* `metrics` settings has been renamed to `monitoring`
```
Comment (apeteri) [inline docs/migration/from-6x.md:39]: An example configuration snippet (before/after) might be worth adding here.

**Review by apeteri — APPROVED** [review]
:+1:, two smaller items posted in the review.

File: docs/migration/from-6x.md (hunk 2 above)
Comment (cmark) [reply docs/migration/from-6x.md:39]: Let's add it later.

**Pull request merged**